### PR TITLE
Improve handling of top level IO and add SDF support

### PR DESCRIPTION
**Pull request by daveshah1**
SDF timing simulation requires some changes to the Yosys cell library that are still being worked on. Furthermore a closed source simulator (e.g. xsim) seems to be required; iverilog's SDF support seems to be too limited, in particular not supporting interconnect delays.

### CODE REVIEW

**Comment by mithro** [comment]
FYI - There is a libsdfparse library at https://github.com/kmurray/libsdfparse if you end up needing one.

File: common/command.cc [hunk 1]
```diff
@@ -149,6 +149,9 @@ po::options_description CommandHandler::getGeneralOptions()
     general.add_options()("freq", po::value<double>(), "set target frequency for design in MHz");
     general.add_options()("timing-allow-fail", "allow timing to fail in design");
     general.add_options()("no-tmdriv", "disable timing-driven placement");
+    general.add_options()("sdf", po::value<std::string>(), "SDF delay back-annotation file to write");
+    general.add_options()("sdf-cvc", "enable tweaks for SDF file compatibility with the CVC simulator");
```
Comment (mithro) [inline common/command.cc]: Just FYI - From what I can see CVC is not **actually** open source. It is more of a much more limited "shared source" style (I was super excited for a minute :-().

Comment (daveshah1) [reply common/command.cc]: It is open source - just a bit hard to download - as far as I can see. But it also has several bugs that stop it from being useful here (very fussy about names in the SDF, incorrect handling of disconnected ports, and possibly others) - and the open source  version is abandoned for all intents and purposes. But there are some GH mirrors like https://github.com/cambridgehackers/open-src-cvc

Comment (mithro) [reply common/command.cc]: From http://www.tachyon-da.com/wp-content/uploads/2016/04/OSS-CVC-MODIFIED-ARTISTIC-LIC.txt

> In particular OSS CVC is intended to use the dual licensing model so any
> commercial redistribution of a modified version, a forked project, a
> commercial product version of OSS CVC, or a version linked with
> commercial closed source software will require other distribution
> arrangements and license with the Copyright Holder.  All version of
> OSS CVC may not be sold by providing access on a cloud and charging 
> a cloud usage fee.

Pretty sure that set of restrictions fails the [Open Source Definition](https://opensource.org/osd-annotated).

**Pull request merged**